File: api-server/src/main/java/com/productresearch/apiserver/domain/source/entity/SourceProduct.java
package com.productresearch.apiserver.domain.source.entity;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "source_product")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Builder
@AllArgsConstructor
public class SourceProduct {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String source;

    @Column(name = "source_product_id")
    private String sourceProductId;

    private String title;
    private BigDecimal price;
    private String url;
    private String seller;

    @Column(name = "raw_json", columnDefinition = "TEXT")
    private String rawJson;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }
}

File: ai-server/app/models/source_product.py
from sqlalchemy import Column, BigInteger, Text, Numeric, TIMESTAMP
from sqlalchemy.sql import func

from app.models.base import Base


class SourceProduct(Base):

    __tablename__ = "source_product"

    id = Column(BigInteger, primary_key=True)

    source = Column(Text)

    source_product_id = Column(Text)

    search_keyword = Column(Text)

    title = Column(Text)

    price = Column(Numeric)

    url = Column(Text)

    seller = Column(Text)

    raw_json = Column(Text)

    created_at = Column(
        TIMESTAMP,
        server_default=func.now()
    )
